packages/unmock-core/src/interfaces.ts

```diff
@@ -85,3 +85,39 @@ export type CreateResponse = (
 // Returns an object (parsed from specification)
 export type RequestToSpec = (sreq: ISerializedRequest) => any;
 export type GeneratedMock = any;
+
+export interface IServiceDefLoader {
+  /**
+   * Asynchronously read service definitions.
+   */
+  load(): Promise<IServiceDef[]>;
+  /**
+   * Synchronously read service definitions.
+   */
+  loadSync(): IServiceDef[];
+}
+
+export interface IServiceDefFile {
+  /**
+   * Basename for the service definition file: for example, `index.yaml`.
+   */
+  basename: string;
+  /**
+   * Contents of the service definition file
+   */
+  contents: string | Buffer;
+}
+
+/**
+ * Input to the service parser. Contains, e.g., the directory name and all available files.
+ */
+export interface IServiceDef {
```

Comment:
The name here is a bit confusing. It's not a service definition at this point, but more of a service def collection?
Maybe `IServiceDefCollection` then?

Reply:
It contains everything that defines a single service (possibly with multiple files), to me `ServiceDefCollection` sounds like something that has to do with multiple services?

Reply:
Ah, to me it read as if it's a collection of services. Alright then! :)